fixing circular import error in providers caused by airflow version check

Lee-W opened the pull request:
We recently worked on testing our project with the latest providers' version, https://github.com/apache/airflow/issues/31322 but encountered a circular import error. Thanks, @ash, for suggesting to me how to solve this on https://github.com/apache/airflow/pull/30994/files#r1197635148

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

Lee-W committed:
fix(amazon): replace airflow.version.version with airflow.__version__ for avoiding circular import error

https://github.com/apache/airflow/pull/30994/files#r1197635148

Lee-W committed:
fix(providers): replace airflow.version.version with airflow.__version__ for avoiding circular import error

Lee-W committed:
fix(providers): fallback to amazon.version.verion if amazon.__version__ does not exist

Lee-W committed:
fix(providers/yandex): revert suspended provider changes

potiuk reviewed (changes requested):
As dicussed in the https://github.com/apache/airflow/pull/30994/files#r1197635148 - we need to also update the JINJA template in `dev/provider_packages` once we figure out the best approach also for 2.4 (just in case I add request changes so that it is not merged accidentally).

Lee-W committed:
fix(dev/provider_packages): update the airflow version import mechanism for providers in the template

Lee-W committed:
fix(providers): replace ImportError with Exception

the original error encounter AttributeError. We might encounter both exception in
different cases

Lee-W committed:
fix(providers): fallback to use importlib if airflow.__version__ cannot be found

Lee-W committed:
fix(providers): catch importlib.metadata ImportError for python 3.7 compatibility

ashb committed:
Use simpler import fallback

ashb committed:
fix OL provider too
